source/game.gamescriptexpression.bmx

```diff
@@ -4,37 +4,627 @@ Import "game.world.worldtime.bmx"
+		'case "population"    Return New SToken( TK_NUMBER, GetGameInformation("stationmap", "population"), params.GetToken(0) )
+		'case "mapname"       Return New SToken( TK_TEXT, GetGameInformation("stationmap", "mapname"), params.GetToken(0) )
+		'case "mapnameshort"  Return New SToken( TK_TEXT, GetGameInformation("stationmap", "mapnameshort"), params.GetToken(0) )
+		case "population"    Return New SToken( TK_NUMBER, GetStationMapCollection().population, params.GetToken(0) )
+		case "mapname"       Return New SToken( TK_TEXT, GetStationMapCollection().GetMapName(), params.GetToken(0) )
+		Case "mapnameshort"  Return New SToken( TK_TEXT, GetStationMapCollection().GetMapISO3166Code(), params.GetToken(0) )
+		default              Return New SToken( TK_ERROR, "(Undefined command ~q"+subCommand+"~q.)", params.GetToken(0) )
+	End Select
+End Function
+
+
+
+'${.persongenerator:***} - context: all
+'${.persongenerator:"firstname":"us":"female"}
+'${.persongenerator:"fullname"}
+Function SEFN_PersonGenerator:SToken(params:STokenGroup Var, context:SScriptExpressionContext)
+	Local command:String = params.GetToken(0).GetValueText()
+	Local subCommand:String = params.GetToken(1).value 'MUST be a string
+	'choose a random country if the country is not defined or no generator
+	'existing for it
+	Local country:String = params.GetToken(2).GetValueText()
+	If country = "" or Not GetPersonGenerator().HasProvider(country)
+		country = GetPersonGenerator().GetRandomCountryCode()
+	EndIf
+	'gender as defined or a random one
+	Local gender:Int = TPersonGenerator.GetGenderFromString( params.GetToken(3).GetValueText() )
+
+	Select subCommand.ToLower()
+		case "name"       Return New SToken( TK_TEXT, GetPersonGenerator().GetFirstName(country, gender), params.GetToken(0) )
+		case "firstname"  Return New SToken( TK_TEXT, GetPersonGenerator().GetFirstName(country, gender), params.GetToken(0) )
+		case "lastname"   Return New SToken( TK_TEXT, GetPersonGenerator().GetLastName(country, gender), params.GetToken(0) )
+		case "fullname"   Return New SToken( TK_TEXT, GetPersonGenerator().GetFirstName(country, gender) + " " + GetPersonGenerator().GetLastName(country, gender), params.GetToken(0) )
+		case "title"      Return New SToken( TK_TEXT, GetPersonGenerator().GetTitle(country, gender), params.GetToken(0) )
+		default           Return New SToken( TK_ERROR, "(PersonGenerator: Undefined command ~q"+subCommand+"~q.)", params.GetToken(0) )
+	End Select
+End Function
+
+
+
+
+'${.locale:"localekey":"optional: language"} - context: all
+Function SEFN_locale:SToken(params:STokenGroup Var, context:SScriptExpressionContext)
+	If params.added >= 1
+		Local key:String = params.GetToken(1).GetValueText()
+		If params.added >= 2
+			Local languageCode:String = params.GetToken(2).value 'MUST be a string
+			Return New SToken( TK_TEXT, GetLocale(key, languageCode), params.GetToken(0) )
+		Else
+			Return New SToken( TK_TEXT, "hallo" + GetLocale(key), params.GetToken(0) )
```

Comment:
Wenn das für das Erkennen nicht aufgelöster Einträge dienen soll, dann lieber "(undefined locale key..."

Reply:
Nee... das bleibt bei "hallo" :blush:

Ernsthaft: wenn Du gerne einen Default-Prefix haben willst, gerne "!Error!/!Warning!/..." oder so. Alternativ kann auch ein Throw wieder eingebaut werden (das war anfangs so beim generellen Parsing schon drin).

Reply:
Ja, Default-Präfix wäre prima !error! ist passender als Warnung, da es ja wirklich ein Fehler bei der Konvertierung oder ein Fehler in der Datenbank ist. throw wäre wohl sauber aber nutzerunfreundlich (debug-Build?).

Übrigens sind nach Hinzufügen der Fehlerpräfixe (die mir aufgefallen sind) noch weitere Probleme aufgefallen!!

Reply:
Erledigt.